lib/openhab/core/dsl/monkey_patch/types/decimal_type.rb

```diff
@@ -10,15 +10,51 @@ class Java::OrgOpenhabCoreLibraryTypes::DecimalType
   # rubocop:enable Style/ClassAndModuleChildren
 
   #
-  # Compare self to other using Java BigDecimal compare method
+  # Compare DecimalType to supplied object
   #
   # @param [Object] other object to compare to
   #
-  # @return [Boolean] True if have the same BigDecimal representation, false otherwise
+  # @return [Integer] -1,0,1 or nil depending on value supplied, nil comparison to supplied object is not possible.
   #
-  def ==(other)
-    return equals(other) unless other.is_a? Integer
+  def <=>(other)
+    logger.trace("#{self.class} #{self} <=> #{other} (#{other.class})")
+    case other
+    when Integer, Float
```

Comment:
why not Numeric instead of these specific types?

Reply:
Ruby Numeric includes complex number but I suppose it won't hurt. However I need to convert the 'other' to string to pass onto Java BigDecimal's constructor.

Reply:
According to https://www.rubyguides.com/2016/07/numbers-in-ruby/ Numeric has several subclasses. Out of those, Java's BigDecimal.new(String) will fail on Rational (because I'm guessing that it won't know "1/2" as a string) and Complex. It would only succeed on Integer, Float and BigDecimal, although I haven't tested this.

Do you think we should stick to Numeric or list the three supported types specifically?

Reply:
Changed to Numeric